akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -414,13 +418,23 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef, joinConfigCompatCh
       joinSeedNodes(newSeedNodes)
     case msg: SubscriptionMessage ⇒
       publisher forward msg
+    case Welcome(from, gossip) ⇒
+      // this can happen if a previous joinSeedNodes was replied to, yet before that we started another joinSeedNodes
+      if (allowWelcomeFromJoiningSeedNodeProcesses.contains(from.address)) {
+        log.info("Accepting Welcome message from previous joinSeedNodeProcess initiated Join, from: [{}]", from)
+        welcome(from.address, from, gossip)
+      } else {
+        log.warning("Received unexpected Welcome message from [{}], the address seems not to be one we attempted to join, ignoring.")
+      }
     case _: Tick ⇒
       if (joinSeedNodesDeadline.exists(_.isOverdue))
         joinSeedNodesWasUnsuccessful()
   }: Actor.Receive).orElse(receiveExitingCompleted)
 
   def tryingToJoin(joinWith: Address, deadline: Option[Deadline]): Actor.Receive = ({
-    case Welcome(from, gossip) ⇒ welcome(joinWith, from, gossip)
+    case Welcome(from, gossip) ⇒
+      require(from.address == joinWith, "joinWith should be the same as the welcome replied from right?")
```

Comment:
don't throw here, it could be different if this is the Welcome from a previous join process and we have started a new that is now trying to join another

Reply:
Okey, was mostly to confirm my understanding that this actually is mostly expected to be the same in normal operation. Removing then